monitoring/ping.go

```diff
@@ -0,0 +1,314 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package monitoring
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/gravitational/satellite/agent/health"
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+
+	"github.com/codahale/hdrhistogram"
+	"github.com/gravitational/trace"
+	"github.com/gravitational/ttlmap"
+	serf "github.com/hashicorp/serf/client"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	// pingCheckerID specifies the check name
+	pingCheckerID = "ping-checker"
+	// slidingWindowSize specifies the number of retained check results
+	slidingWindowSize = 10
```

Comment:
I think this window size is too small to indicate anything and might produce a lot of false positives, let's increase to a 100 at least.

Reply:
10 was the proposed number initially. It seems that the check runs every 30 seconds so 100 would take a lot of time to change the 95 percentile in case things go bad, I think.
I guess it's a tradeoff between being reactive and dealing with false positives by tweaking the threshold and the slidingWindow.
I'll put 100 for now but it seems a little too  wide as a window IMO.